lib/backend/k8s/syncer.go

```diff
@@ -67,12 +68,50 @@ func (syn *kubeSyncer) Start() {
 	go syn.readFromKubernetesAPI()
 }
 
+// sendUpdates sends updates to the callback and updates the resource
+// tracker.
+func (syn *kubeSyncer) sendUpdates(updates []api.Update) {
+	// Send to the callback and update the tracker.
+	syn.callbacks.OnUpdates(updates)
+	syn.updateTracker(updates)
+}
+
+// updateTracker updates the global object tracker with the given update.
+// updateTracker should be called after sending a update to the OnUpdates callback.
+func (syn *kubeSyncer) updateTracker(updates []api.Update) {
+	for _, upd := range updates {
+		if upd.UpdateType == api.UpdateTypeKVDeleted {
+			log.Debugf("Delete from tracker: %+v", upd.KVPair.Key)
+			delete(syn.tracker, upd.KVPair.Key)
+		} else {
+			log.Debugf("Update tracker: %+v: %+v", upd.KVPair.Key, upd.KVPair.Revision)
+			syn.tracker[upd.KVPair.Key] = upd.KVPair.Revision
+		}
+	}
+}
+
+// isUpdated returns true if the given KVPair is either new or has changed
+// since it was last seen, and false if it hasn't changed.
```

Comment:
Method doesn't look to be used, but I don't think the comment matches the method.  You're returning true if the key is new or if it has the same value as before; false iff the key is known and the value is different.